samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -111,27 +113,20 @@ namespace Microsoft.Quantum.Samples.SudokuGrover {
     : (Bool, Int[]) {
         // for size = 4x4 grid
         let bitsPerColor = size == 9 ? 4 | 2;
-        mutable oracle = ApplyVertexColoringOracle(numVertices, bitsPerColor, emptySquareEdges, startingNumberConstraints, _, _);
-        if (size == 9) {
-            // Although we could use ApplyVertexColoringOracle for 9x9, we would
-            // have to add restrictions on each color to not allow colors 8 
-            // through 15. This could be achieved by adding these to 
-            // startNumberConstraints. However, this did not scale well with 
-            // the simulator, and so instead we use 
-            // ApplyVertexColoringOracle4Bit9Color which has the 9 color 
-            // restriction built in.
-            set oracle = ApplyVertexColoringOracle4Bit9Color(numVertices, emptySquareEdges, startingNumberConstraints, _, _);
-        } elif (size != 4) {
+        let oracle = ApplyVertexColoringOracle(numVertices, bitsPerColor, startingNumberConstraints, _, _);
+        let statePrep = PrepareSearchStatesSuperposition(numVertices, bitsPerColor, startingNumberConstraints, _);
+        let searchSpaceSize = SearchSpaceSize(numVertices, bitsPerColor, startingNumberConstraints);
+        if (size != 4 and size != 9) {
```

Comment:
I'd move this check to be the first thing in the operation. Nothing bad will happen from defining operations for other sizes, but logically we want to do this check before we proceed with the solution